File: test-fixtures/languages/elixir/complex/advanced.ex
defmodule MyApp.Advanced do
  @moduledoc """
  Complex Elixir fixture exercising pattern matching, pipes, try/rescue,
  case, for comprehensions.
  """

  def find_user(users, id) do
    case Map.get(users, id) do
      nil -> nil
      user -> user
    end
  end

  def load_all_users do
    try do
      for i <- 0..9, into: %{} do
        {i, "user_#{i}"}
      end
    rescue
      e ->
        IO.puts("load failed: #{inspect(e)}")
        %{}
    end
  end

  def escape_sql(input) do
    String.replace(input, "'", "''")
  end

  def dispatch_token(token) do
    cond do
      token == "" -> :skip
      String.starts_with?(token, "admin_") -> run_admin(token)
      true -> run_user(token)
    end
  end

  def process_batch(tokens) do
    Enum.each(tokens, fn token -> dispatch_token(token) end)
  end

  def run_admin(token) do
    System.cmd("sh", ["-c", "admin-task " <> token])
  end

  def run_user(token) do
    System.cmd("sh", ["-c", "user-task " <> token])
  end
end

File: test-fixtures/languages/dart/complex/advanced.dart
// Complex multi-construct Dart fixture exercising branches, loops,
// try/catch, async/await, and classes.
import 'dart:io';
import 'dart:async';

class UserRepository {
  Map<int, String> users = {};

  String? findUser(int id) {
    if (users.containsKey(id)) {
      return users[id];
    }
    return null;
  }

  Future<bool> loadAllUsers() async {
    try {
      for (var i = 0; i < 10; i++) {
        users[i] = "user_$i";
      }
      return true;
    } catch (e) {
      print("load failed: $e");
      return false;
    }
  }
}

String escapeSQL(String input) {
  return input.replaceAll("'", "''");
}

void dispatchToken(String token) {
  if (token.isEmpty) {
    return;
  }
  if (token.startsWith("admin_")) {
    runAdmin(token);
  } else {
    runUser(token);
  }
}

void processBatch(List<String> tokens) {
  for (var token in tokens) {
    dispatchToken(token);
  }
}

void runAdmin(String token) {
  Process.runSync('sh', ['-c', 'admin-task $token']);
}

void runUser(String token) {
  Process.runSync('sh', ['-c', 'user-task $token']);
}

Future<String> asyncFetch(String key) async {
  await Future.delayed(Duration(milliseconds: 10));
  return "value:$key";
}
